api/app/Policies/PoolPolicy.php

```diff
@@ -32,10 +32,14 @@ public function viewAny(User $user)
      * @param  \App\Models\Pool  $pool
      * @return \Illuminate\Auth\Access\Response|bool
      */
-    public function view(User $user, Pool $pool)
+    public function view(?User $user, Pool $pool)
     {
-        // Guests and Base Users both have permission to view-any-publishedPoolAdvertisement
+        // If not logged in, we still want to allow access to return published pools.
+        if (!$user) {
+            return true;
+        }
```

Comment:
It was in the wrong place before, but seems clearer to exit the policy with a return when we get to the place where we expect user to be available, rather than peppering the assumption throughout with new syntax